### ISSUE TITLE
mis-detection of cached packages

### ISSUE THREAD

**Issue by aronatkins** [rstudio/packrat#335]
Assume that your packrat cache is in some location and has a package `shiny` installed.

```
cd packrat_cache/3.2.5/v2/library
mv shiny shiny-old
```

Now, there is a `shiny-old/28d6903a44dc53bd4823fa43ccdc08e5/shiny/` directory inside the cache. This appears to be sufficient for a subsequent `packrat::restore` to attempt to use. However, it creates a symlink from `packrat/lib/.../shiny` to `packrat_cache/3.2.5/v2/library/shiny/28d6903a44dc53bd4823fa43ccdc08e5/shiny`.

In other words, packrat finds the matching `28d6903a44dc53bd4823fa43ccdc08e5` hash and does not consider that the other pieces of the hierarchy might not match.

This is a likely scenario only when manually manipulating the cache.

**Comment by kevinushey** [comment]
Should be resolved now; please re-open later if you encounter issues.
